### PR TITLE
Allow GC of arguments during microtask execution

### PR DESCRIPTION
**Pull request by Nevay**

### CODE REVIEW

File: src/EventLoop/Internal/AbstractDriver.php [hunk 1]
```diff
@@ -418,7 +418,8 @@ private function invokeMicrotasks(): void
             [$callback, $args] = $this->microtaskQueue->dequeue();
 
             try {
-                $callback(...$args);
+                // Clear $args to allow garbage collection
+                $callback(...$args, ...($args = []));
```
Comment (WyriHaximus) [inline src/EventLoop/Internal/AbstractDriver.php:422]: So how does this work? You first pass all the elements of the `$args` array into the argument list for `$callback`, and then you overwrite  `$args` with an empty array and also "push" that to the same argument list?

Comment (Nevay) [reply src/EventLoop/Internal/AbstractDriver.php:422]: Yes, it overwrites the `$args` array (and pushes zero arguments to the argument list) after pushing the original arguments and before invoking the callback.

**Comment by bwoebi** [comment]
~The alternative implementation would be `\call_user_func_array(...$this->microtaskQueue->dequeue());` directly. Which looks cleaner to me than this clever hack :-D~

... Though, it might be that this doesn't work here, as cufa() stores its own CoW copy of its args until its end...

**Comment by trowski** [comment]
@Nevay Thanks for finding this and for the creative solution. Seems we rarely pass arguments, let alone instantiating objects in the argument list, so this hadn't come up. We appreciate the real-world testing to find these cases!

File: src/EventLoop/Internal/AbstractDriver.php (hunk 1 above)
Comment (WyriHaximus) [reply src/EventLoop/Internal/AbstractDriver.php:422]: Near trick, it's a tad confusing when you first read the line tbh.

Comment (HLeithner) [reply src/EventLoop/Internal/AbstractDriver.php:422]: A better documentation what happens here would be good especially for people like me that doesn't see what's happening at the first glance and why it's needed.

**Pull request merged**